awsenv/replacer_test.go

```diff
@@ -133,3 +184,87 @@ type mockParamsGetter func(context.Context, []string) (map[string]string, error)
 func (f mockParamsGetter) GetParams(ctx context.Context, paths []string) (map[string]string, error) {
 	return f(ctx, paths)
 }
+
+func TestFilterPaths(t *testing.T) {
+	tests := []struct {
+		prefix string
+		input  map[string]string
+		want   []string
+	}{
+		{
+			prefix: "",
+			input:  nil,
+			want:   []string{},
+		},
+		{
+			prefix: "awsenv:",
+			input:  map[string]string{"X": "1", "Y": "pre:/y", "Z": "awsenv:/z"},
+			want:   []string{"/z"},
+		},
+		{
+			prefix: "pre:",
+			input:  map[string]string{"X": "1", "Y": "pre:/y", "Z": "awsenv:/z"},
+			want:   []string{"/y"},
+		},
+	}
+
+	for _, test := range tests {
+		r := Replacer{prefix: test.prefix}
+		got, want := r.filterPaths(test.input), test.want
+		if !reflect.DeepEqual(got, want) {
```

Comment:
Why are we using `reflect` for comparison here when we use packages like `/stretchr/testify` for assertions in other tests?

Reply:
I didn't change that from the original code.